### PR TITLE
feature/default-switch-packetfence-standard

### LINKED ISSUE #5490: RFC: use "PacketFence::Standard" in place of "Generic" for default Switch Group
**Describe the bug**
Currently, default switch group still use `Generic` as default switch (old switch module). We should move to `PacketFence::Standard` (template module). That's way, all new switches created from default switch group will use `PacketFence::Standard` (template module).

Is there is some limitations ?

I saw two functions in `Generic.pm` related to CLI access to switches (https://github.com/inverse-inc/packetfence/blob/5ba7bf9d8863837d8c81013bf443237fd8691a70/lib/pf/Switch/Generic.pm) and I wonder if we can handle those things in `PacketFence::Standard`.

From my point of view, that's the case.

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Description
===========
Set the default switch type to Packetfence::Standard

NEWS file entries
=======================

Enhancements
-------------
* Set the default switch type to Packetfence::Standard
* Create a PacketFence::SNMP switch to force reevaluate access using SNMP

Issue
=====
fixes #5490

Delete branch after merge
=========================
NO

UPGRADE file entries
=========================

Set the switch type to Generic if it was not previously defined.

/usr/local/pf/addons/upgrade/to-10.2-default-switch-packetfence-standard.pl

Checklist
=========
- [ ] Document the feature
- [x] Add unit tests
- [ ] Add acceptance tests (TestLink)

### CODE REVIEW

**Review by nqb — CHANGES_REQUESTED** [review]
There is an ambiguity for `addons/upgrade/to-10.2-default-switch-packetfence-standard.pl` script.

Name of script doesn't match its description: "Rename Generic switch type to Packetfence::SNMP" and its actions: script is trying to set Generic value for switches.

**Comment by jrouzierinverse** [comment]
Fixed the verbage

File: addons/upgrade/to-10.2-default-switch-packetfence-standard.pl [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+to-10.2-default-switch-packetfence-standard.pl
+
+=cut
+
+=head1 DESCRIPTION
+
+Add Generic type where needed
+
+=cut
```
Comment (nqb) [inline addons/upgrade/to-10.2-default-switch-packetfence-standard.pl:13]: If purpose of this PR is to "Set the default switch type to Packetfence::Standard", title and description needs to be adjust.

Comment (nqb) [reply addons/upgrade/to-10.2-default-switch-packetfence-standard.pl:13]: Also description of PR (Upgrade section) need to be edited.

**Review by nqb — COMMENTED** [review]
Should we not remove Generic switch module too ?

**Review by nqb — CHANGES_REQUESTED** [review]
If we want to replace `Generic` switch module by `PacketFence::Standard` switch module, we should first merge #5708, rebase and add `cliAuthorizeRead` and `cliAuthorizeWrite` scopes to `PacketFence::Standard` switch module based on what we do in Generic switch module.

I can take care of this change.

**Comment by jrouzierinverse** [comment]
Let's remove the Generic switch in 11.0.

**Comment by jrouzierinverse** [comment]
There is no need to add cliAuthorizeRead and cliAuthorizeWrite scopes to PacketFence::Standard.
As it has the same behavior of the Generic module if none are defined.

File: addons/upgrade/to-10.2-default-switch-packetfence-standard.pl (hunk 1 above)
Comment (nqb) [reply addons/upgrade/to-10.2-default-switch-packetfence-standard.pl:13]: Purpose of this script is to set Generic type to switch which don't have any type.

**Pull request merged**